storage/innobase/row/row0ins.cc

```diff
@@ -2711,6 +2711,20 @@ row_ins_clust_index_entry_low(
 
 	DBUG_EXECUTE_IF("row_ins_row_level", goto skip_bulk_insert;);
 
+	/* Ignore the bulk insert operation for temporary,
+	versioning table. In case of foreign key constraint,
+	InnoDB has to check the constraint validity for
+	each row. But bulk insert operation doesn't
+	check foreign constraint. Skip bulk insert
+	operation if the table contains foreign key relation */
+	if (index->table->skip_alter_undo
+	    && !index->table->is_temporary()
+	    && !index->table->versioned()
+	    && index->table->foreign_set.empty()) {
+		trx_start_if_not_started(trx, true);
+		goto bulk_insert;
+	}
+
 	if (!(flags & BTR_NO_UNDO_LOG_FLAG)
```

Comment:
Can we avoid adding a new bunch of unlikely conditions at the start of a busy function (inserting a record into a clustered index)? Can we relax the subsequent `if` statement condition so that `goto bulk_insert` will not be needed? That condition is written in such a way that the least likely or easiest-to-evaluate conditions are checked first.